ehrapy/preprocessing/_encoding.py

```diff
@@ -85,29 +78,42 @@ def encode(
         if "original" not in adata.layers.keys():
             adata.layers["original"] = adata.X.copy()
 
-        # autodetect categorical values, which could lead to more categoricals
+        # autodetect categorical values based on feature types stored in adata.var[FEATURE_TYPE_KEY]
         if autodetect:
-            if "var_to_encoding" in adata.uns.keys():
-                logger.warning(
-                    "The current AnnData object has been already encoded. Returning original AnnData object!"
-                )
-                return adata
-            categoricals_names = _get_var_indices_for_type(adata, NON_NUMERIC_TAG)
+            categoricals_names = _get_var_indices_for_type(adata, CATEGORICAL_TAG)
+
+            if "encoding_mode" in adata.var.keys():
+                if adata.var["encoding_mode"].isnull().values.any():
+                    not_encoded_features = adata.var["encoding_mode"].isna().index
+                    categoricals_names = [cat for cat in categoricals_names if cat in not_encoded_features]
+                else:
+                    logger.warning(
+                        "The current AnnData object has been already encoded. Returning original AnnData object!"
+                    )
+                    return adata
+
+            # filter out categorical columns, that are already stored numerically
+            df_adata = anndata_to_df(adata)
+            categoricals_names = [
+                feat
+                for feat in categoricals_names
+                if not np.all(df_adata[feat].apply(type).isin([int, float, complex]))
+            ]
 
             # no columns were detected, that would require an encoding (e.g. non-numerical columns)
             if not categoricals_names:
                 logger.warning("Detected no columns that need to be encoded. Leaving passed AnnData object unchanged.")
                 return adata
             # copy uns so it can be used in encoding process without mutating the original anndata object
```

Comment:
the comment in L107 says "copy uns so it can be ...", now would be "copy obs so it can be ..." to avoid confusion